tests/karate/src/test/java/be/utils/JsonConverter.java

```diff
@@ -0,0 +1,160 @@
+package be.utils;
+
+import com.google.crypto.tink.PublicKeySign;
+import com.google.crypto.tink.subtle.Ed25519Sign;
+import com.intuit.karate.Json;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.Base64;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class JsonConverter {
+
+  private String senderPk = "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=";
+  private String privateKeyHex = "d257820c1a249652572974fbda9b27a85e54605551c6773504d0d2858d392874";
+  private String signatureForced = "ONylxgHA9cbsB_lwdfbn3iyzRd4aTpJhBMnvEKhmJF_niE_pUHdmjxDXjEwFyvo5WiH1NZXWyXG27SYEpkasCA==";
+  private String messageIdForced = "";
+
+  public Json fromMapToJson(Map<String, String> map) {
+    return Json.of(map);
+  }
+
+  /**
+   * Produces the base64 variant of the json file passed as argument
+   */
+  public String convertJsonToBase64(Json json) {
+    String stringJson = json.toString();
+    byte[] jsonBytes = stringJson.getBytes();
+    Base64.Encoder encoder = Base64.getEncoder();
+    return encoder.encodeToString(jsonBytes);
+  }
+
+  /**
+   * Produces a valid Json representation of a message given the message data, the id of the message
+   * and the channel where the message is supposed to be sent
+   */
+  public Json publishМessageFromData(String stringData, int id, String channel) {
+    Json messageData = Json.of(stringData);
+    Map<String, Object> messageJson = new LinkedHashMap<>();
+
+    messageJson.put("method", "publish");
+    messageJson.put("id", id);
+
+    Map<String, Object> paramsPart = constructParamsField(channel, stringData);
+
+    messageJson.put("params", paramsPart);
+    messageJson.put("jsonrpc", "2.0");
+
+    return Json.of(messageJson);
+  }
+
+  public Map<String, Object> constructMessageField(String stringData, boolean forcedMessageId) {
+    Map<String, Object> messagePart = new LinkedHashMap<>();
+    Json messageData = Json.of(stringData);
+    String messageDataBase64 = convertJsonToBase64(messageData);
+    messagePart.put("data", messageDataBase64);
+    messagePart.put("sender", senderPk);
+
+    String signature = constructSignature(stringData);
+    messagePart.put("signature", signature);
+    String messageId = hash(messageDataBase64.getBytes(), signature.getBytes());
+
+    if (forcedMessageId && !messageIdForced.isEmpty()) {
+      messageId = this.messageIdForced;
+      messageIdForced = "";
+    }
```

Comment:
I am not a fan of this method, as we talked about in the last meeting, I think this should be removed completely.